nilearn/decoding/decoder.py

```diff
@@ -106,41 +106,130 @@ def _check_param_grid(estimator, X, y, param_grid=None):
+        the affine is considered the same for all.
+
+    y: array or list of shape (n_samples)
+        The dependent variable (age, sex, IQ, yes/no, etc.).
+        Target variable to predict. Must have exactly as many elements as
+        3D images in niimg.
+
+    Returns
+    -------
+    param_grid: dict of str to sequence, or sequence of such. Sensible default
+    dict has size 1 for linear models.
+    """
+    param_grid = {}
+
+    # define loss function
+    if isinstance(estimator, LogisticRegression):
+        loss = "log"
+    elif isinstance(estimator, (LinearSVC, RidgeCV, RidgeClassifierCV, SVR)):
+        loss = "squared_hinge"
+    elif isinstance(estimator, (DummyClassifier, DummyRegressor)):
+        if estimator.strategy in ["constant"]:
+            message = (
+                "Dummy classification implemented only for strategies"
+                ' "most_frequent", "prior", "stratified"'
+            )
+            raise NotImplementedError(message)
+    else:
+        raise ValueError(
+            "Invalid estimator. The supported estimators are:"
+            f" {list(SUPPORTED_ESTIMATORS.keys())}"
+        )
+    # define sensible default for different types of estimators
+    if hasattr(estimator, "penalty") and (estimator.penalty == "l1"):
+        min_c = l1_min_c(X, y, loss=loss)
+    else:
+        min_c = 0.5
+
+    if isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+        param_grid["alphas"] = [np.geomspace(1e-3, 1e4, 8)]
+    elif not isinstance(estimator, (DummyClassifier, DummyRegressor)):
+        param_grid["C"] = np.array([2, 20, 200]) * min_c
+    else:
         param_grid = {}
-        # define loss function
-        if isinstance(estimator, LogisticRegression):
-            loss = "log"
-        elif isinstance(
-            estimator, (LinearSVC, RidgeCV, RidgeClassifierCV, SVR)
+
+    return param_grid
+
+
+def _wrap_param_grid(param_grid, param_name):
+    """Wrap a parameter's sequence of values with an outer list.
+
+    This can be
+    desirable for models featuring built-in cross-validation, as it would leave
+    it to the model's internal (optimized) cross-validation to loop over
+    hyperparameter values. Does nothing if the parameter is already wrapped.
+
+    Parameters
+    ----------
+    param_grid : dict of str to sequence, or sequence of such
+        The parameter grid to wrap, as a dictionary mapping estimator
+        parameters to sequences of allowed values.
+    param_name : str
+        Name of parameter whose sequence of values should be wrapped
+
+    Returns
+    -------
+    param_grid_wrapped: dict of str to sequence, or sequence of such
+        The updated parameter grid
+    """
+    if param_grid is None:
+        return param_grid
+
+    # param_grid can be either a dict or a sequence of dicts
+    # we make sure that it is a sequence we can loop over
+    input_is_dict = isinstance(param_grid, dict)
+    if input_is_dict:
+        param_grids = [param_grid]
```

Comment:
I find it helps readability to avoid names that differ by just one letter, like `param_grid` and `param_grids`. here as `param_grid` itself can be a sequence in some cases, you could write `param_grid = [param_grid]`, or if you would rather have a different name it could be for example `all_param_grids` (in which case you might want to make `new_param_grid` plural, too)